pkg/registry/apis/datasource/plugincontext.go

```diff
@@ -31,6 +31,10 @@ type PluginDatasourceProvider interface {
 	GetInstanceSettings(ctx context.Context, uid string) (*backend.DataSourceInstanceSettings, error)
 }
 
+type ScopedPluginDatasourceProvider interface {
+	GetDatasourceProvider(pluginJson plugins.JSONData) PluginDatasourceProvider
```

Comment:
https://github.com/grafana/grafana/pull/84753 -- hardcoded testdata, this adds a layer of abstraction so it will work for any datasource.